bpf-gpl/tc.c

```diff
@@ -586,9 +586,9 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 	}
 
 	/* We are possibly past (D)NAT, but that is ok, we need to let the IP
-	 * stack do the RPF check on the source, dest is not importatnt.
+	 * stack do the RPF check on the source, dest is not important.
 	 */
-	if (CALI_F_TO_HOST && ct_result_rpf_failed(state.ct_result.rc)) {
```

Comment:
The tiny benefit of the `CALI_F_TO_HOST &&` is that it is compiled out for other programs. Probably makes little difference in runtime and the code is simpler without it.

Reply:
I'm hoping the compiler is smart enough to spot that fib is already set to false at the top so this conditional is a no-op.